Code/Framework/AzCore/Tests/Debug/PerformanceCollectorTests.cpp

```diff
@@ -0,0 +1,215 @@
+            auto argsObj = jsonDoc[i]["args"].GetObject();
+            ASSERT_TRUE(argsObj.HasMember(AZ::Debug::PerformanceCollector::AVG.data()));
+            ASSERT_TRUE(argsObj.HasMember(AZ::Debug::PerformanceCollector::MIN.data()));
+            ASSERT_TRUE(argsObj.HasMember(AZ::Debug::PerformanceCollector::MAX.data()));
+            ASSERT_TRUE(argsObj.HasMember(AZ::Debug::PerformanceCollector::STDEV.data()));
+            ASSERT_TRUE(argsObj.HasMember(AZ::Debug::PerformanceCollector::SAMPLE_COUNT.data()));
+
+            // Remark, here we call GetDouble() and later cast to unsigned int, because
+            // GetUint() returns weird numbers when reading a floating point value.
+            auto avgAsDouble = argsObj[AZ::Debug::PerformanceCollector::AVG.data()].GetDouble();
+            EXPECT_EQ(jsonDoc[i]["dur"].GetUint(), aznumeric_cast<unsigned int>(avgAsDouble));
+
+            auto sampleCount = argsObj[AZ::Debug::PerformanceCollector::SAMPLE_COUNT.data()].GetUint();
+            if (paramName == PerfParam1)
+            {
+                // Periodic events work in deltas. The first frame is just for collecting
+                // the Begin time, so there's no delta. For subsequent events there's
+                // now previous data to subtract from and that's why we check for (FrameCount - 1).
+                EXPECT_EQ(sampleCount, frameCountPerCaptureBatch - 1);
+            }
+            else
+            {
+                EXPECT_EQ(sampleCount, frameCountPerCaptureBatch);
+            }
+        }
+
+    }
+
+    TEST_F(PerformanceCollectorTest, CreatePerformanceCollector_CollectPerformance_ValidateAllSamplesOutput)
+    {
+        // In this test we use the PerformanceCollector to collect all data for three parameters.
+        // the data is NOT statistically summarize so each time RecordPeriodicEvent() or
+        // the AZ::Debug::ScopeDuration is used, a unique row (aka json object) will be added to the output json.
+
+        const AZStd::string LogCategory("PerformanceCollectorTest");
+        constexpr AZStd::string_view PerfParam1("param1");
+        constexpr AZStd::string_view PerfParam2("param2");
+        constexpr AZStd::string_view PerfParam3("param3");
+
+        // Define control arguments that will define how much data to collect.
+        // In real life these parameters are mirrored by CVARs.
+        const AZ::u32 numberOfCaptureBatches = 3;
+        const AZ::Debug::PerformanceCollector::DataLogType dataLogType = AZ::Debug::PerformanceCollector::DataLogType::LogAllSamples;
+        const AZ::u32 waitTimePerCaptureBatch = 0; //0 seconds
+        const AZ::u32 frameCountPerCaptureBatch = 10;
+
+        AZ::u32 pendingBatchCount = numberOfCaptureBatches;
+        auto onCompleteCallback = [&](AZ::u32 batchCount) {
+            pendingBatchCount = batchCount;
+        };
+        AZStd::vector<AZStd::string_view> paramList = { PerfParam1, PerfParam2, PerfParam3 };
```

Comment:
To avoid allocating memory, an `AZStd::array<AZStd::string_view, 3>` or an `AZStd::fixed_vector<AZStd::string_view, 3>` can be used.
```suggestion
        auto paramList = AZStd::to_array<AZStd::string_view>({ PerfParam1, PerfParam2, PerfParam3 });
```

Reply:
Changed to AZStd::to_array<AZStd::string_view>